#!/usr/bin/env bash

set -euxo pipefail

WORK_DIR=~/STR_Expansions_Example_Workflow


# Required input (default choices for example workflow listed below):
# - bam files generated by ExpansionHunter
# - database of repeats used by ExpansionHunter

# Optional input
# - genotype files generated by ExpansionHunter (vcf / json)
# - reference genome
# - output prefix and directory

BAMS=`ls $WORK_DIR/results/ExpansionHunter/*_realigned.bam`

REPEATDB_EH=$WORK_DIR/repeats/ExpansionHunter_known_hg38.json

REFERENCE=$WORK_DIR/minibam/Homo_sapiens_assembly38.fasta

OUTPUT_DIR=$WORK_DIR/results/GraphAlignmentViewer


# Run GraphAlignmentViewer
mkdir -p $OUTPUT_DIR

for BAM in $BAMS
do
    BASE=$(basename $BAM "_realigned.bam")
    mkdir -p $OUTPUT_DIR/$BASE
    
    python3 $WORK_DIR/tools/GraphAlignmentViewer/GraphAlignmentViewer.py \
        --read_align $BAM \
        --variant_catalog $REPEATDB_EH \
        --gt_file $WORK_DIR/results/ExpansionHunter/$BASE.vcf \
        --reference_fasta $REFERENCE \
        --output_prefix $BASE \
        --output_dir $OUTPUT_DIR/$BASE
done


